hail/python/hail/methods/impex.py

```diff
@@ -423,8 +423,23 @@ def export_vcf(dataset, output, append_to_header=None, parallel=None, metadata=N
         dictionary should be structured.
 
     """
-
     require_row_key_variant(dataset, 'export_vcf')
+    row_fields_used = {'rsid', 'info', 'filters', 'qual'}
+
+    fields_dropped = []
+    for f in dataset.globals:
+        fields_dropped.append((f, 'global'))
+    for f in dataset.col_value:
+        fields_dropped.append((f, 'column'))
+    for f in dataset.row_value:
+        if not f in row_fields_used:
+            fields_dropped.append((f, 'row'))
+
+    if fields_dropped:
+        ignored_str = ''.join(f'\n    {f!r} ({axis})' for f, axis in fields_dropped)
```

Comment:
what is `f!r` in the format string?

Reply:
this is the same as `{repr(f)}`. We should print the repr here because we need to escape the field name.